File: InventoryManagement.Core/Entities/Quotation.cs
using InventoryManagement.Core.Common;
using InventoryManagement.Core.Enums;

namespace InventoryManagement.Core.Entities;

public class Quotation : BaseEntity
{
    public string QuotationNumber { get; set; } = string.Empty;
    public DateTime QuotationDate { get; set; }
    public DateTime ValidUntil { get; set; }
    public int CustomerId { get; set; }
    public QuotationStatus Status { get; set; } = QuotationStatus.Draft;
    public decimal TotalAmount { get; set; }
    public decimal? TaxAmount { get; set; }
    public decimal? DiscountAmount { get; set; }
    public decimal GrandTotal { get; set; }
    public string? Notes { get; set; }
    public string? TermsAndConditions { get; set; }

    // Navigation Properties
    public Customer Customer { get; set; } = null!;
    public ICollection<QuotationItem> QuotationItems { get; set; } = new List<QuotationItem>();
}

File: InventoryManagement.UI/src/app/core/models/quotation.ts
export enum QuotationStatus {
  Draft = 'Draft',
  Sent = 'Sent',
  Accepted = 'Accepted',
  Rejected = 'Rejected',
  Expired = 'Expired'
}

export interface Quotation {
  id: number;
  quotationNumber: string;
  quotationDate: Date;
  validUntil: Date;
  customerId: number;
  customerName: string;
  status: QuotationStatus;
  totalAmount: number;
  taxAmount?: number;
  discountAmount?: number;
  grandTotal: number;
  notes?: string;
  termsAndConditions?: string;
  createdAt: Date;
  items: QuotationItem[];
}

export interface QuotationItem {
  id: number;
  itemId: number;
  itemName: string;
  itemCode: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
  discountPercent?: number;
  discountAmount?: number;
  netAmount: number;
  description?: string;
}

export interface CreateQuotationRequest {
  quotationDate: Date;
  validUntil: Date;
  customerId: number;
  notes?: string;
  termsAndConditions?: string;
  items: CreateQuotationItemRequest[];
}

export interface CreateQuotationItemRequest {
  itemId: number;
  quantity: number;
  unitPrice: number;
  discountPercent?: number;
  description?: string;
}

export interface UpdateQuotationRequest {
  quotationDate: Date;
  validUntil: Date;
  customerId: number;
  notes?: string;
  termsAndConditions?: string;
  items: CreateQuotationItemRequest[];
}

export interface PagedQuotationResult {
  items: Quotation[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
  hasPreviousPage: boolean;
  hasNextPage: boolean;
}